osquery/filesystem/fileops.h

```diff
@@ -426,4 +426,8 @@ boost::filesystem::path getSystemRoot();
  * @return osquery::Status
  */
 Status platformLstat(const std::string& path, struct stat& d_stat);
+
+#ifdef __APPLE__
```

Comment:
My recommendation is to drop this check, allow the function to exist on all platforms.